api/v1alpha1/statuscheck_types.go

```diff
@@ -0,0 +1,230 @@
+
+import (
+	"net/http"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +chaos-mesh:base
+type StatusCheck struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the behavior of a status check
+	Spec StatusCheckSpec `json:"spec"`
+
+	// +optional
+	// Most recently observed status of status check
+	Status StatusCheckStatus `json:"status"`
+}
+
+type StatusCheckMode string
+
+const (
+	// StatusCheckSynchronous means the status check will exit
+	// immediately after success or failure.
+	StatusCheckSynchronous StatusCheckMode = "Synchronous"
+	// StatusCheckContinuous means the status check will continue to
+	// execute until the duration is exceeded or the status check fails.
+	StatusCheckContinuous StatusCheckMode = "Continuous"
+)
+
+type StatusCheckType string
+
+const (
+	TypeHTTP StatusCheckType = "HTTP"
+)
+
+type StatusCheckSpec struct {
+	// Mode defines the execution mode of the status check.
+	// Support type: Synchronous / Continuous
+	// +optional
+	// +kubebuilder:validation:Enum=Synchronous;Continuous
+	Mode StatusCheckMode `json:"mode,omitempty"`
+
+	// Type defines the specific status check type.
+	// Support type: HTTP
+	// +kubebuilder:default=HTTP
+	// +kubebuilder:validation:Enum=HTTP
+	Type StatusCheckType `json:"type"`
+
+	// Duration defines the duration of the whole status check if the
+	// number of failed execution does not exceed the failure threshold.
+	// Duration is available to both `Synchronous` and `Continuous` mode.
+	// A duration string is a possibly signed sequence of
+	// decimal numbers, each with optional fraction and a unit suffix,
+	// such as "300ms", "-1.5h" or "2h45m".
+	// Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+	// +optional
+	Duration *string `json:"duration,omitempty" webhook:"Duration"`
+
+	// TimeoutSeconds defines the number of seconds after which
+	// an execution of status check times out.
+	// +optional
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=1
+	TimeoutSeconds int `json:"timeoutSeconds,omitempty"`
+
+	// IntervalSeconds defines how often (in seconds) to perform
+	// an execution of status check.
+	// +optional
+	// +kubebuilder:default=10
+	// +kubebuilder:validation:Minimum=1
+	IntervalSeconds int `json:"intervalSeconds,omitempty"`
+
+	// FailureThreshold defines when an execution of status check
+	// fails, it will try FailureThreshold times before giving up.
+	// +optional
+	// +kubebuilder:default=3
+	// +kubebuilder:validation:Minimum=1
+	FailureThreshold int `json:"failureThreshold,omitempty"`
+
+	// SuccessThreshold defines the minimum consecutive successes
+	// for the status check to be considered successful.
+	// +optional
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=1
+	SuccessThreshold int `json:"successThreshold,omitempty"`
+
```

Comment:
Maybe we could use similar descriptions/comments for both FailureThreshold and SuccessThreshold?

And it seems SuccessThreshold would only make sense with `mode: Synchronous`?

Reply:
Updated.